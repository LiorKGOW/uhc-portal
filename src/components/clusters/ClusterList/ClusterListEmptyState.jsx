// ClusterListEmptyState is the empty state (no clusters) for ClusterList
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import {
  Button, Grid, Row, Col, EmptyState,
} from 'patternfly-react';


function ClusterListEmptyState(props) {
  const { showCreationForm, showOCPCreationForm } = props;
  return (
    <EmptyState className="full-page-blank-slate">
      <Grid fluid>
        <Row>
          <EmptyState.Title>
            No Clusters Exist
          </EmptyState.Title>
          <EmptyState.Info>
            There are no clusters to display. Create a cluster to get started.
          </EmptyState.Info>
        </Row>
        <Row className="cluster-list-emptystate-options">
          <Col md={4}>
            <h2>Self-Managed (Manual Install) Cluster</h2>
            <p>
              Create a self-managed cluster (OCP)
              to install OpenShift and manage it yourself.
              More information and better text should go here.
            </p>
            <EmptyState.Action>
              <Link to="/clusters/install">
                <Button bsStyle="primary" bsSize="large">
                  Create Self-Managed (Manual) Cluster
                </Button>
              </Link>
            </EmptyState.Action>
          </Col>
          <Col md={4}>
            <h2>Self-Managed (Auto-Install) Cluster</h2>
            <p>
              Create a self-managed cluster (OCP)
              to provision the cluster on Amazon Web Services and manage it yourself.
              More information and better text should go here.
            </p>
            <EmptyState.Action>
              <Button bsSize="large" onClick={showOCPCreationForm}>
                Create Self-Managed (Auto) Cluster
              </Button>
            </EmptyState.Action>
          </Col>
          <Col md={4}>
            <h2>Red Hat-Managed Cluster</h2>
            <p>
              Create a Red Hat-managed cluster (OSD),
              to provision the cluster on Amazon Web Services.
              More information on this flow and why users would take this flow.
            </p>
            <EmptyState.Action>
              <Button bsSize="large" onClick={showCreationForm}>
                Create Red Hat-Managed Cluster
              </Button>
            </EmptyState.Action>
          </Col>
        </Row>
      </Grid>
    </EmptyState>
  );
}

ClusterListEmptyState.propTypes = {
  showCreationForm: PropTypes.func.isRequired,
  showOCPCreationForm: PropTypes.func.isRequired,
};


export default ClusterListEmptyState;
