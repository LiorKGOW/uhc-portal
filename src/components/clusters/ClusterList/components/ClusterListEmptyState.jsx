// ClusterListEmptyState is the empty state (no clusters) for ClusterList
import React from 'react';
import PropTypes from 'prop-types';
import {
  Button, Row, Col, EmptyState, Card, CardGrid, CardTitle, CardBody,
} from 'patternfly-react';

function ClusterListEmptyState(props) {
  const { showCreationForm, hasQuota } = props;

  const createManaged = (
    <EmptyState.Action>
      <Button bsStyle="primary" bsSize="large" onClick={showCreationForm}>
        Create Red Hat-Managed Cluster
      </Button>
    </EmptyState.Action>
  );

  const createSelfManaged = (
    <EmptyState.Action>
      <Button bsStyle="primary" bsSize="large" href={APP_EMBEDDED ? '/openshift/install' : '/install'} target="_blank">
        Create Self-Installed Cluster
      </Button>
    </EmptyState.Action>
  );

  return (
    <EmptyState className="full-page-blank-slate">
      <Row>
        <EmptyState.Title>
          No Clusters Exist
        </EmptyState.Title>
        <EmptyState.Info>
          There are no clusters to display. Create a cluster to get started.
        </EmptyState.Info>
      </Row>
      <CardGrid matchHeight>
        <Row className="cluster-list-emptystate-options">
          {hasQuota && (
            <React.Fragment>
              <Col md={4} mdOffset={2}>
                <Card className="cluster-list-emptystate-preferred" accented matchHeight>
                  <CardBody>
                    <p className="cluster-list-emptystate-preferred-string">Preferred method</p>
                    <CardTitle>Red Hat-Managed Cluster</CardTitle>
                    <p>
                  Create a Red Hat-managed cluster (OSD),
                  to provision the cluster on Amazon Web Services.
                    </p>
                    {createManaged}
                  </CardBody>
                </Card>
              </Col>
            </React.Fragment>)}
          <Col md={4} mdOffset={hasQuota ? 0 : 4}>
            <Card matchHeight>
              <CardBody>
                <CardTitle>Self-Installed Cluster</CardTitle>
                <p>
                  Install an OCP cluster manually, and manage it yourself.
                  Once installation is complete the cluster will get automatically
                  registered to the Cluster Manager.
                </p>
                {createSelfManaged}
              </CardBody>
            </Card>
          </Col>
        </Row>
      </CardGrid>
    </EmptyState>
  );
}

ClusterListEmptyState.propTypes = {
  showCreationForm: PropTypes.func.isRequired,
  hasQuota: PropTypes.bool,
};

export default ClusterListEmptyState;
