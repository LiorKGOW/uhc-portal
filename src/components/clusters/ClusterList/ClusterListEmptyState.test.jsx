import React from 'react';
import { shallow } from 'enzyme';

import ClusterListEmptyState from './ClusterListEmptyState';

describe('<ClusterListEmptyState />', () => {
  let onClickFunc;
  let wrapper;
  beforeEach(() => {
    onClickFunc = jest.fn();
    wrapper = shallow(<ClusterListEmptyState showCreationForm={onClickFunc} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls showCreationForm when needed', () => {
    wrapper.find('Button').at(1).simulate('click');
    expect(onClickFunc).toBeCalled();
  });
});
