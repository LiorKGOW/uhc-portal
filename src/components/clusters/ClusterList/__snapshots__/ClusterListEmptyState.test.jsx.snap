// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClusterListEmptyState /> renders correctly 1`] = `
<EmptyState
  className="full-page-blank-slate"
>
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <EmptyStateTitle
        className=""
      >
        No Clusters Exist
      </EmptyStateTitle>
      <EmptyStateInfo
        className=""
      >
        There are no clusters to display. Create a cluster to get started.
      </EmptyStateInfo>
    </Row>
    <Row
      bsClass="row"
      className="cluster-list-emptystate-options"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={3}
        mdOffset={3}
      >
        <h2>
          Self Managed Cluster
        </h2>
        <p>
          Create a self-managed cluster (OCP) to install OpenShift and manage it yourself. More information and better text should go here.
        </p>
        <EmptyStateAction
          className=""
          secondary={false}
        >
          <Link
            replace={false}
            to="/clusters/install"
          >
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsSize="large"
              bsStyle="primary"
              disabled={false}
            >
              Create Self Managed Cluster
            </Button>
          </Link>
        </EmptyStateAction>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={3}
      >
        <h2>
          Red Hat Managed Cluster
        </h2>
        <p>
          Create a Red Hat-managed cluster (OSD), to provision the cluster on Amazon Web Services. More information on this flow and why users would take this flow.
        </p>
        <EmptyStateAction
          className=""
          secondary={false}
        >
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsSize="large"
            bsStyle="default"
            disabled={false}
            onClick={[MockFunction]}
          >
            Create Red Hat Managed Cluster
          </Button>
        </EmptyStateAction>
      </Col>
    </Row>
  </Grid>
</EmptyState>
`;
